Makefile

```diff
@@ -3,58 +3,78 @@ GOARCH ?= $(shell go env GOARCH)
 GOOS_GOARCH := $(GOOS)_$(GOARCH)
 GOOS_GOARCH_NATIVE := $(shell go env GOHOSTOS)_$(shell go env GOHOSTARCH)
 LIBZSTD_NAME := libzstd_$(GOOS_GOARCH).a
-ZSTD_VERSION ?= v1.5.7
-ZIG_BUILDER_IMAGE=euantorano/zig:0.10.1
-BUILDER_IMAGE := local/builder_musl:2.0.0-$(shell echo $(ZIG_BUILDER_IMAGE) | tr : _ | tr / _)-1
+ZSTD_VERSION ?= v1.5.7-kernel
+ZIG_BUILDER_IMAGE := euantorano/zig:0.10.1
 
-.PHONY: libzstd.a $(LIBZSTD_NAME)
+# Detect available container runtime
+CONTAINER_RUNTIME := $(shell \
+	if command -v docker >/dev/null 2>&1; then \
+		echo "docker"; \
+	elif command -v nerdctl >/dev/null 2>&1; then \
+		echo "nerdctl"; \
+	elif command -v podman >/dev/null 2>&1; then \
+		echo "podman"; \
+	else \
+		echo "none"; \
+	fi)
+
+# Show which runtime is being used
+$(info Using container runtime: $(CONTAINER_RUNTIME))
+
+# Parallel compilation flags
+JOBS := $(shell nproc 2>/dev/null || sysctl -n hw.ncpu 2>/dev/null || echo 4)
+MAKEFLAGS += -j$(JOBS)
+
+.PHONY: libzstd.a $(LIBZSTD_NAME) test test-no-lto bench bench-no-lto
 
 libzstd.a: $(LIBZSTD_NAME)
 $(LIBZSTD_NAME):
 ifeq ($(GOOS_GOARCH),$(GOOS_GOARCH_NATIVE))
 	rm -f $(LIBZSTD_NAME)
-	cd zstd/lib && ZSTD_LEGACY_SUPPORT=0 MOREFLAGS=$(MOREFLAGS) $(MAKE) clean libzstd.a
+	cd zstd/lib && ZSTD_LEGACY_SUPPORT=0 AR="gcc-ar" ARFLAGS="rcs" MOREFLAGS="-DZSTD_MULTITHREAD=1 -O3 -flto=1 $(MOREFLAGS)" LDFLAGS="-flto=1 -fuse-linker-plugin" $(MAKE) clean libzstd.a
 	mv zstd/lib/libzstd.a $(LIBZSTD_NAME)
 else ifeq ($(GOOS_GOARCH),linux_amd64)
-	TARGET=x86_64-linux GOARCH=amd64 GOOS=linux $(MAKE) package-arch
+	TARGET=x86_64-linux GOARCH=amd64 GOOS=linux ARCH_FLAGS="-mcpu=x86_64+sse4_2+avx2+bmi2" $(MAKE) package-arch
```

Comment:
Likewise these `ARCH_FLAGS` are performance related and raises the AMD64 microarchitecture level to v3 which could be considered a breaking change? (_support should be fairly widespread these days but you'll find for example in Docker images that v1 is the default unless explicitly set_)

These are ideal for you to opt-in, but when explicit like this you are forcefully reducing compatibility. That's often a tradeoff when distributing a project to a wider audience than yourself.

This sort of thing should be handled better and via a separate PR.